cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,296 @@
+
+    def _adapt_from_base(self):
+        """
+        For each base section, find the matching one in target configuration (or create it if missing).
+
+        To mark correspondence, assign the same section to the base section as the target section
+        """
+        for section_key in self.base_config.get_section_keys():
+            for _, base_section in self.base_config.get_sections(section_key).items():
+                target_section = self._get_target_section(base_section)
+                # Assign to base config's section the same label as the target section
+                base_section.label = target_section.label
+                LOGGER.debug(
+                    "Matched base section {0} {1} with target section {0} {2}".format(
+                        section_key, base_section.label, target_section.label
+                    )
+                )
+
+    def _adapt_from_target(self):
+        """Create a base section for each target section that is still not matched."""
+        for section_key in self.target_config.get_section_keys():
+            for _, target_section in self.target_config.get_sections(section_key).items():
+                base_section = self._get_base_section(target_section)
+                LOGGER.debug(
+                    "Matched target section {0} {1} with base section {0} {2}".format(
+                        section_key, target_section.label, base_section.label
+                    )
+                )
+
+    def _compare(self):
+        """
+        Compare the target config to the source one.
+
+        All sections in both configurations are compared one by one. This is made possible by the adaptation phase.
+        The last compared section is "cluster". This is compared only after the mock sections created during the
+        adaptation phase are removed, in order to make differences in settings parameters (like ebs_settings) appear
+        correctly.
+        """
+        # First, compare all sections except "cluster".
+        for section_key in self.target_config.get_section_keys():
+            if section_key != "cluster":
+                for _, target_section in self.target_config.get_sections(section_key).items():
+                    base_section = self.base_config.get_section(target_section.key, target_section.label)
+                    self._compare_section(base_section, target_section)
+
+        # Then, remove all mock sections. Settings_parameters will be refresh after this.
+        self._remove_mock_sections(self.base_config)
+        self._remove_mock_sections(self.target_config)
+
+        # Finally, compare cluster sections.
+        self._compare_section(
+            self.base_config.get_section("cluster"), self.target_config.get_section("cluster"),
+        )
+
+    def _compare_section(self, base_section, target_section):
+        for _, param in target_section.params.items():
+            base_value = base_section.get_param(param.key).value if base_section else None
```

Comment:
We don't need to verify if base_section exist because we created them in the adapt phase